kvdb/sqlite/driver.go

```diff
@@ -80,7 +83,29 @@ func openDBDriver(args ...interface{}) (walletdb.DB, error) {
 		return nil, err
 	}
 
-	return NewSqliteBackend(ctx, config, dbPath, filename, prefix)
+	// Check if the database file already exists. The walletdb.Open function
+	// is meant to open existing databases, but the underlying driver might
+	// create one if it doesn't exist. We need to know if we are creating
+	// it here to set the initial pragmas correctly (like auto_vacuum).
+	fullPath := filepath.Join(dbPath, filename)
```

Comment:
We'll need to do the same thing for `sqldb.NewSqliteStore(...)` https://github.com/lightningnetwork/lnd/blob/6a3845b79da835071bf03085007b4a6e76c009d3/sqldb/sqlite.go#L52 which is used if the user requested to run lnd with native schema.

Reply:
Ah good catch. Added in the latest commit.